k8s/core/configmap/configmap_lock_v2.go

```diff
@@ -442,3 +435,20 @@ func (c *configMap) checkLockTimeout(holdTimeout time.Duration, startTime time.T
 		}
 	}
 }
+
+// check whether the requester is the owner of the lock in configmap but has no goroutine
+// to refresh the lock expiration time
+func (c *configMap) ifRequesterIsLockOwnerWithoutGoroutine(requester, owner, key string) bool {
```

Comment:
Let's put more data on comment to give context:
We want to avoid locks being re-entrant. At the same time make sure the owner re-aquires the lock if there has been a restart. For that reason we check if the refresh routine is running. On a restart, the routine would have been cancelled.